sleap/gui/commands.py

```
@@ -3470,13 +3481,178 @@ def do_action(context: CommandContext, params: dict):
+        # Write frames to the video
+        for frame_idx in range(*frame_range):
+            try:
+                frame = video.get_frame(frame_idx)
+
+                # Convert grayscale to BGR if necessary
+                if frame.ndim == 2:  # Grayscale frames
+                    frame = cv2.cvtColor(frame, cv2.COLOR_GRAY2BGR)
+
+                writer.add_frame(frame, bgr=True)
+            except KeyError:
+                raise KeyError(f"Failed to load frame {frame_idx} from video.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)